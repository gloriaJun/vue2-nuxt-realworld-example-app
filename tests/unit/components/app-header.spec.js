import { shallowMount } from '@vue/test-utils';
import AppHeader from '@/components/app-header.vue';

const siteName = 'siteName';

describe('Header.vue', () => {

  const build = () => {
    const wrapper = shallowMount(AppHeader, {
      stubs: ['nuxt-link'],
      propsData: { siteName: siteName },
    });

    return {
      wrapper,
    }
  }

  it('should match the snapshot', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('check brand link', () => {
    const { wrapper } = build();

    wrapper.find('.navbar-brand').trigger('click');
    expect(window.location.href).toBe(`http://${window.location.host}/`);
  });

  it('check brand name', () => {
    const { wrapper } = build();
    expect(wrapper.find('.navbar-brand').text()).toBe(siteName);
  });

  it('check home link', () => {
    const { wrapper } = build();

    const el = wrapper.findAll('li.nav-item');
    expect(el.length).toBe(4);
  });

})
