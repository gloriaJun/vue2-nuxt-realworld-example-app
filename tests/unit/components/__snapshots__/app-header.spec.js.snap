// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue should match the snapshot 1`] = `
<nav class="navbar navbar-light">
  <div class="container">
    <nuxt-link-stub to="/" class="navbar-brand">
      siteName
    </nuxt-link-stub>
    <ul class="nav navbar-nav pull-xs-right">
      <li class="nav-item">
        <nuxt-link-stub to="/" class="nav-link">
          Home
        </nuxt-link-stub>
      </li>
      <li class="nav-item">
        <nuxt-link-stub to="/post" class="nav-link"><i class="ion-compose"></i>&nbsp;New Post
        </nuxt-link-stub>
      </li>
      <li class="nav-item">
        <nuxt-link-stub to="/settings" class="nav-link"><i class="ion-gear-a"></i>&nbsp;Settings
        </nuxt-link-stub>
      </li>
      <li class="nav-item">
        <nuxt-link-stub to="/register" class="nav-link">
          Sign up
        </nuxt-link-stub>
      </li>
    </ul>
  </div>
</nav>
`;
